metadata/src/main/java/io/kaoto/backend/metadata/parser/JsonProcessFile.java

```diff
@@ -0,0 +1,15 @@
+package io.kaoto.backend.metadata.parser;
+
+import io.kaoto.backend.model.Metadata;
+
+import java.io.File;
+
+public abstract class JsonProcessFile<T extends Metadata>
+        extends ProcessFile<T> {
+
+    @Override
+    protected boolean isDesiredType(final File file) {
```

Comment:
Is there no other json files in the camel jar that may be read here? 

Maybe we should check the path of the file too, that it contains some folders so at least we know the JSON file is inside the proper component modules.

Reply:
I think there is some way to go with that, as camel components are always generated in a generated/metadata folder if I can recall correctly. I'll check up this later.

Reply:
I have improved it, removed subdirectories without Camel Components and have a naive validation checking for the looked structure.

Reply:
Where? I can't find it.

Reply:
Found it!